### PR TITLE
migration guide for ColoredBox/container optimization

### PR DESCRIPTION
**Pull request by dnfield**
flutter/flutter#50979

### CODE REVIEW

File: src/docs/release/breaking-changes/container-color.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+* [`BoxDecoration`][]
+
+Relevant issues:
+* [Issue 9672][]
+* [Issue 28753][]
+
+Relevant PRs:
+* [Colored box and container optimization #50979][]
+
+{% comment %}
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/container-color.md]: Please remove the comment markers and the text that introduces them. thx

File: src/docs/release/breaking-changes/container-color.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+It is very common to use the `Container` widget as follows:
+
+```dart
+return Container(color: Colors.red);
+```
+
+Previously, this code would result in a widget heirarchy that used a
+`BoxDecoration` to actually paint the background color. The `BoxDecoration`
+widget covers many cases other than just painting a background color, and is
+not as efficient as the new `ColoredBox` widget which only paints a background
```
[the remark this answers is not in the document]
Comment (dnfield) [reply src/docs/release/breaking-changes/container-color.md]: Done

File: src/docs/release/breaking-changes/container-color.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+color.
+
+Widget tests that wanted to assert based on the color of a container in the
+widget tree would previously have to find the `BoxDecoration` to actually get
+the color of the container. Now, they will be able to check the `color` property
```
[the remark this answers is not in the document]
Comment (dnfield) [reply src/docs/release/breaking-changes/container-color.md]: Done

File: src/docs/release/breaking-changes/container-color.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  // If your test needed to work directly with the BoxDecoration, it should
+  // instead look for the ColoredBox, e.g.:
+  expect(find.byType(BoxDecoration), findsNothing);
+  expect(find.byType(ColoredBox), findsOneWidget);
+});
+```
+
+## Timeline
+
+This change was introduced in v1.TBD
```
[the remark this answers is not in the document]
Comment (dnfield) [reply src/docs/release/breaking-changes/container-color.md]: Right - I'll update this once the change has actually landed and I have a version number

File: src/docs/release/breaking-changes/container-color.md (hunk 1 above)
Comment (dnfield) [reply src/docs/release/breaking-changes/container-color.md]: Done

File: src/docs/release/breaking-changes/container-color.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (sfshaza2) [reply src/docs/release/breaking-changes/container-color.md]: thx!

**Comment by dnfield** [comment]
Resolved the merge conflict and updated the version number.  @sfshaza2  PTAL

**Pull request merged**